src/main.cpp

```diff
@@ -763,6 +765,29 @@ void setup()
   new Characteristic::Version();
   homeSpan.setPairCallback(pairCallback);
   homeSpan.setWifiCallback(wifiCallback);
+  mqtt.connected_callback = [] {
+    char bufferpub[256]; // create a local char array to store the JSON string
+    char device[100]; // create a local char array to store the device JSON string
+    Serial.println("MQTT connected");
+  if (strcmp(DISCOVERY, "1") == 0) {
+      json payload;
+      payload["topic"] = MQTT_AUTH_TOPIC;
+      payload["value_template"] = "{{ value_json.uid }}";
+      json device; // create a JSON object for the device
+      device["name"] = NAME; // assign the name value
+      device["identifiers"] = {NAME}; // assign the identifiers array
+      payload["device"] = device; // assign the device object to the payload object
+      std::string bufferpub = payload.dump(-1, ' ', false, json::error_handler_t::strict); // use dump instead of dump_to
+      mqtt.publish(("homeassistant/tag/" NAME "/" NAME "-rfid/config"), bufferpub.c_str(), true, 1);     
+      payload = json();
+      payload["topic"] = MQTT_AUTH_TOPIC;
+      payload["value_template"] = "{{ value_json.issuerId }}";
```

Comment:
not sure how HASS Tags work but would it make sense to also announce a tag with the endpoint id? or maybe replace issuerId altogether?

Reply:
scratch that, think it's a bit late for me 😬 💤, i think it should actually be included since it would give more automation options.